tests/e2e/file_volume_statefulsets.go

```diff
@@ -508,4 +508,175 @@ var _ = ginkgo.Describe("[csi-file-vanilla] File Volume statefulset", func() {
 		gomega.Expect(len(ssPodsAfterScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
 			"Number of Pods in the statefulset should match with number of replicas")
 	})
+
+	/*
+	   Test performs following operations
+
+	   Steps
+	   1. Create a storage class.
+	   2. Create nginx service.
+	   3. Create nginx statefulsets with 3 replicas.
+	   4. Wait until all Pods are ready and PVCs are bounded with PV.
+	   5. Restart CSI dameonset pods
+	   6. Scale down statefulsets to 2 replicas.
+	   7. Scale up statefulsets to 3 replicas.
+	   8. Scale down statefulsets to 0 replicas and delete all pods.
+	   9. Delete all PVCs from the tests namespace.
+	   10. Delete the storage class.
+	*/
+	ginkgo.It("Statefulset with file volume testing with CSI daemonset restart", func() {
+		curtime := time.Now().Unix()
+		randomValue := rand.Int()
+		val := strconv.FormatInt(int64(randomValue), 10)
+		val = string(val[1:3])
+		curtimestring := strconv.FormatInt(curtime, 10)
+		scName := "nginx-sc-default-" + curtimestring + val
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass for Statefulset")
+		scParameters[scParamFsType] = nfs4FSType
+		scSpec := getVSphereStorageClassSpec(scName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+		statefulset := GetStatefulSetFromManifest(namespace)
+		ginkgo.By("Creating statefulset")
+		statefulset.Spec.VolumeClaimTemplates[len(statefulset.Spec.VolumeClaimTemplates)-1].Spec.AccessModes[0] =
+			v1.ReadWriteMany
+		statefulset.Spec.VolumeClaimTemplates[len(statefulset.Spec.VolumeClaimTemplates)-1].
+			Annotations["volume.beta.kubernetes.io/storage-class"] = scName
+		CreateStatefulSet(namespace, statefulset, client)
```

Comment:
add a defer block to delete statefulset

Reply:
It is already taken care in the after-each method.